Audiocodes VoiceAI Connect documentation (#11815)
RasaHQ/rasa: docs/docs/connectors/audioodes-voiceai-connect.mdx

---
id: audiocodes-voiceai-connect
sidebar_label: Audiocodes VoiceAI Connect
title: Audiocodes VoiceAI Connect
description: Build a Rasa Voice Bot on Audiocodes VoiceAI Connect
---

import RasaProLabel from '@theme/RasaProLabel';
import RasaProBanner from '@theme/RasaProBanner';

<RasaProLabel/>

<RasaProBanner/>

Use this channel to connect your Rasa assistant to [Audiocodes VoiceAI connect](https://www.audiocodes.com/solutions-products/voiceai/voiceai-connect).
## Getting Credentials

To get credentials, create a bot on the [VoiceAI connect portal](https://voiceaiconnect.audiocodes.io/).

1. Select **Bots** in the left side bar.
2. Click on the **+** sign to create a new bot.

> 3. Select **Rasa** in the platforms.

Reviewer: ```suggestion
3. Select **Rasa** as the Bot Framework
```

4. Then set the bot URL a choose a token value.

## Setting credentials

The token value chosen above will be used in the credentials:

```yaml
rasa_plus.channels.audiocodes.AudiocodesInput:
    token: <token>
```

You can also specify optional parameters:

| Parameter       | Default value    | Description                                                                                                                 |
| --------------- | ---------------- | --------------------------------------------------------------------------------------------------------------------------- |
| `token`         | No default value | The token to authenticate calls between your Rasa assistant and VoiceAI connect                                             |
| `use_websocket` | `true`           | If `true`, Rasa will send messages through a web socket. If set to `false`, Rasa will send messages through http API calls. |

> | `keep_alive`    | 120              | For each ongoing conversation, VoiceAI will periodically verifies the conversation is still active on the Rasa side.        |

Reviewer: ```suggestion
| `keep_alive`    | 120              | For each ongoing conversation, VoiceAI Connect will periodically verify the conversation is still active on the Rasa side.        |
```

| `DTMF`  | `vaig_event_DTMF`  | VoiceAI will send this intent when receiving a DTMF tone (i.e user presses digit on the keyboard of the phone). The digit(s) sent will be passed in the `value` entity                                                                                                                                                                     |

The general pattern is that for every `event` sent, the bot will receive the `vaig_event_<event>` intent, with context information in entities.

Here is a simple rule to send a greeting message when a call to the bot is initiated:

```yaml
- rule: New call
  steps:
      - intent: vaig_event_start
      - action: utter_greet
```

Check the [VoiceAI Connect documentation](https://techdocs.audiocodes.com/voice-ai-connect/#VAIG_Combined/voiceai_connect.htm?TocPath=VoiceAI%2520Connect%257C_____0) for an exhaustive list of events.

### Configuring calls

> You can send events from Rasa to VoiceAI to make changes to the current call configuration.

Reviewer: ```suggestion
You can send events from Rasa to VoiceAI Connect to make changes to the current call configuration.
```

For example, you might want to receive a notification when the user stays silent for more than 5 seconds, or you might need to customize how DTMF digits are sent by VoiceAI Connect.

Call configuration events are sent with custom messages and are specific to the current conversation (sometimes to a message).
Which means they must be part of your stories or rules so the same behaviour is applied to all conversations.

> There is a lot of different events which are exhaustively documented in the [VoiceAI Connect documentation](https://techdocs.audiocodes.com/voice-ai-connect/#VAIG_Combined/voiceai_connect.htm?TocPath=VoiceAI%2520Connect%257C_____0).

Reviewer: ```suggestion
All the supported events are exhaustively documented in the [VoiceAI Connect documentation](https://techdocs.audiocodes.com/voice-ai-connect/#VAIG_Combined/voiceai_connect.htm?TocPath=VoiceAI%2520Connect%257C_____0).
```

> We will look at one example ehere so you understand how these work and how to use them.

Reviewer: ```suggestion
We will look at one example here to illustrate the use of custom messages and events.
```

#### Example: changing a pin code

In this example we create a flow to allow a user to change a pin code.

```yaml
- rule: Set pin code
  steps:
  - intent: set_pin_code # User says "I want to change my pin code"
  - action: utter_config_no_user_input # Send the noUserInput configuration event
  - action: utter_config_dtmf_pin_code # Send the DTMF format configuration event
  - action: pin_code_form # A standard Rasa form to collect the pin code from the user
  - ...
```

In the domain, we can add the `utter_config...` responses:

[`noUserInput` event](https://techdocs.audiocodes.com/voice-ai-connect/#VAIG_Combined/inactivity-detection.htm?TocPath=Bot%2520integration%257CReceiving%2520notifications%257C_____3)
```yaml
  utter_config_no_user_input:
  - custom:
      type: event
      name: config
      sessionParams:

>         userNoInputTimeoutMS: 5000 # If user stays silent for 5 seconds or more, the notification will be sent
>         userNoInputRetries: 2 # If you want to allow for more than one notification during a call
>         userNoInputSendEvent: true # Enable the noUserInput notification

Reviewer 2: ```suggestion
        # If user stays silent for 5 seconds or more, the notification will be sent
        userNoInputTimeoutMS: 5000
        # If you want to allow for more than one notification during a call
        userNoInputRetries: 2
        # Enable the noUserInput notification
        userNoInputSendEvent: true
```
